tools/idf.py

```diff
@@ -340,6 +340,9 @@ def _get_esptool_args(args):
     with open(os.path.join(args.build_dir, "flasher_args.json")) as f:
         flasher_args = json.load(f)
 
+    extra_esptool_args = flasher_args["extra_esptool_args"]
+    result += ["--before", extra_esptool_args["before"]]
+
     extra_esptool_args = flasher_args["extra_esptool_args"]
```

Comment:
This extra line of `extra_esptool_args = flasher_args["extra_esptool_args"]` should be removed.

Reply:
I removed extra_esptool_args = flasher_args["extra_esptool_args"] online:343.
Please check it.